src/axom/sidre/interface/c_fortran/wrapfsidre.F

```diff
@@ -1625,80 +1625,80 @@ function c_group_create_group_bufferify(self, path, Lpath, &
             use iso_c_binding, only : C_CHAR, C_INT, C_PTR
-            import :: SHROUD_group_capsule
+            import :: SIDRE_SHROUD_group_capsule
             implicit none
-            type(SHROUD_group_capsule), intent(IN) :: self
+            type(SIDRE_SHROUD_group_capsule), intent(IN) :: self
             character(kind=C_CHAR), intent(IN) :: path(*)
             integer(C_INT), value, intent(IN) :: Lpath
-            type(SHROUD_group_capsule), intent(OUT) :: SHT_crv
+            type(SIDRE_SHROUD_group_capsule), intent(OUT) :: SHT_crv
             type(C_PTR) SHT_rv
         end function c_group_create_group_bufferify
 
         subroutine c_group_destroy_group_name(self, path) &
                 bind(C, name="SIDRE_Group_destroy_group_name")
             use iso_c_binding, only : C_CHAR
-            import :: SHROUD_group_capsule
+            import :: SIDRE_SHROUD_group_capsule
             implicit none
-            type(SHROUD_group_capsule), intent(IN) :: self
+            type(SIDRE_SHROUD_group_capsule), intent(IN) :: self
             character(kind=C_CHAR), intent(IN) :: path(*)
         end subroutine c_group_destroy_group_name
 
         subroutine c_group_destroy_group_name_bufferify(self, path, &
                 Lpath) &
                 bind(C, name="SIDRE_Group_destroy_group_name_bufferify")
             use iso_c_binding, only : C_CHAR, C_INT
-            import :: SHROUD_group_capsule
+            import :: SIDRE_SHROUD_group_capsule
             implicit none
-            type(SHROUD_group_capsule), intent(IN) :: self
+            type(SIDRE_SHROUD_group_capsule), intent(IN) :: self
             character(kind=C_CHAR), intent(IN) :: path(*)
             integer(C_INT), value, intent(IN) :: Lpath
         end subroutine c_group_destroy_group_name_bufferify
 
         subroutine c_group_destroy_group_index(self, idx) &
                 bind(C, name="SIDRE_Group_destroy_group_index")
             use iso_c_binding, only : C_INT64_T
-            import :: SHROUD_group_capsule
+            import :: SIDRE_SHROUD_group_capsule
             implicit none
-            type(SHROUD_group_capsule), intent(IN) :: self
+            type(SIDRE_SHROUD_group_capsule), intent(IN) :: self
             integer(C_INT64_T), value, intent(IN) :: idx
         end subroutine c_group_destroy_group_index
 
         function c_group_move_group(self, grp, SHT_crv) &
                 result(SHT_rv) &
                 bind(C, name="SIDRE_Group_move_group")
             use iso_c_binding, only : C_PTR
-            import :: SHROUD_group_capsule
+            import :: SIDRE_SHROUD_group_capsule
             implicit none
-            type(SHROUD_group_capsule), intent(IN) :: self
-            type(SHROUD_group_capsule), intent(IN) :: grp
-            type(SHROUD_group_capsule), intent(OUT) :: SHT_crv
+            type(SIDRE_SHROUD_group_capsule), intent(IN) :: self
+            type(SIDRE_SHROUD_group_capsule), intent(INOUT) :: grp
```

Comment:
Was this expected to change?

Reply:
Yes. When a derived type is passed to Fortran with intent(OUT) the compiler will clear all of the members since the routine is declaring that it will assign them (intent(OUT)).  By changing to intent(INOUT), the compiler will not clear the derived type and will not complain if the subroutine assigns to individual members of the struct.  intent(IN) says that no members will change.